runtime/src/lib.rs

```diff
@@ -198,7 +199,56 @@ impl Drivers {
             #[cfg(feature = "fips_self_test")]
             self_test_status: SelfTestStatus::Idle,
             cert_chain,
-        })
+        };
+
+        if reset_reason != ResetReason::ColdReset {
+            let dpe = &drivers.persistent_data.get().dpe;
+            let tci_type = u32::from_be_bytes(*b"RTJM");
+            let root_idx = dpe
+                .get_root_idx(tci_type, locality)
+                .map_err(|_| CaliptraError::RUNTIME_DPE_VALIDATION_FAILED)?;
+
+            // Ensure context/TCI tree is well-formed (all nodes reachable from root and no cycles)
+            if dpe.validate_context_tree(root_idx) {
```

Comment:
There's a lot of code underneath this `if` block. Could the condition be inverted and we just bail early on failure?

Reply:
Done.